src/Tools/dotnet-monitor/Auth/AuthOptions.cs

```diff
@@ -14,10 +16,20 @@ internal sealed class AuthOptions : IAuthOptions
 
         public bool EnableKeyAuth => (KeyAuthenticationMode == KeyAuthenticationMode.StoredKey) ||
                                      (KeyAuthenticationMode == KeyAuthenticationMode.TemporaryKey);
+        
+        public byte[] GeneratedKey { get; }
 
         public AuthOptions(KeyAuthenticationMode mode)
         {
             KeyAuthenticationMode = mode;
+            
+            if (mode == KeyAuthenticationMode.TemporaryKey)
+            {
+                byte[] newKey = new byte[32];
+                RNGCryptoServiceProvider rngCsp = new RNGCryptoServiceProvider();
```

Comment:
What is the difference between RNGCryptoServiceProvider and RandomNumberGenerator? Note that GenerateApiKey command uses the latter. Should that be changed to use RNGCryptoServiceProvider?

Reply:
(It would also be nice to actually have the same code generate the key in both cases, should we ever change how we generate it in the first place)

Reply:
I'm apparently wrong on this. In .NET framework, the standard is to always use `new RNGCryptoServiceProvider()` because `RandomNumberGenerator` can be a non-cryptographically secure algorithm. In .NET core the standard is to use `RandomNumberGenerator.Create()` and it will only ever give you a `RNGCryptoServiceProvider`.